caddyhttp/proxy/upstream.go

```diff
@@ -416,9 +423,21 @@ func (u *staticUpstream) healthCheck() {
 			}
 
 			if r, err := u.HealthCheck.Client.Do(req); err == nil {
-				io.Copy(ioutil.Discard, r.Body)
-				r.Body.Close()
-				unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
+				if u.HealthCheck.ContentString != "" {
+					b, err := ioutil.ReadAll(r.Body)
```

Comment:
Hey Matt thanks for the detailed review and sorry for taking so long to get back. 

ReadAll() does make me cringe a little. But, like you, I was thinking health check endpoints are a controlled resource. I figured, whoever sets one up, even if its dynamically generated, will probably only have one line of content.  A bufio.NewScanner could be used instead, to bail early. 

Was the discussion in #general. I missed it.